db/migrate/20230811081854_populate_fees_precise_coupons_amount_cents.rb

```diff
@@ -63,16 +69,19 @@ def change
               base_amount_cents = fees.sum(:amount_cents)
               next if base_amount_cents.zero?
 
-              fee.precise_coupons_amount_cents = (credit.amount_cents * fee.amount_cents).fdiv(base_amount_cents)
+              fee.precise_coupons_amount_cents += (credit.amount_cents * fee.amount_cents).fdiv(base_amount_cents)
+              fee.save!
             end
           else
-            fees = credit.invoice.fees
+            fees = credit.invoice.fees.where.not(id: fees_id)
 
             fees.find_each do |fee|
-              base_amount_cents = fees.sum(:amount_cents)
+              # NOTE: When applying coupons without limitations, the base of computation is the remaining amount of all fees.
```

Comment:
Layout/LineLength: Line is too long. [125/120] (https://rubystyle.guide#max-line-length)